Source/JavaScriptCore/dfg/DFGJITCompiler.cpp

```diff
@@ -800,6 +809,20 @@ std::tuple<CompileTimeStructureStubInfo, JITCompiler::LinkableConstant> JITCompi
     return std::tuple { stubInfo, LinkableConstant() };
 }
 
+std::tuple<CompileTimeCallLinkInfo, JITCompiler::LinkableConstant> JITCompiler::addCallLinkInfo(CodeOrigin codeOrigin)
+{
+    if (m_graph.m_plan.isUnlinked()) {
+        void* unlinkedCallLinkInfoIndex = bitwise_cast<void*>(static_cast<uintptr_t>(m_unlinkedCallLinkInfos.size()));
+        UnlinkedCallLinkInfo* callLinkInfo = &m_unlinkedCallLinkInfos.alloc();
+        callLinkInfo->codeOrigin = codeOrigin;
+        LinkerIR::Constant callLinkInfoIndex = addToConstantPool(LinkerIR::Type::CallLinkInfo, unlinkedCallLinkInfoIndex);
+        return std::tuple { callLinkInfo, LinkableConstant(callLinkInfoIndex) };
```

Comment:
So this adds a callLinkInfo to m_unlinkedCallLinkInfos, but why can't it add it to m_jitCode directly and avoid the move above?

Reply:
Because we would like to use FixedVector in JITCode to save memory as much as possible. JITCode is alive so long, but JITCompiler dies quickly. We allocate the backing vector in a faster way in JITCompiler and when finally building JITCode, copy them with the smallest size to keep them small.